web/src/engine/websites/MangaTown.ts

```diff
@@ -19,7 +40,7 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override async FetchPages(chapter: Chapter): Promise<Page[]> {
-        const data = await FetchCSS<HTMLOptionElement>(new Request(new URL(chapter.Identifier, this.URI)), 'div.manga_read_footer div.page_select select option');
-        return data.filter(option => !/featured\.html$/.test(option.value)).map(option => new Page(this, chapter, new URL(option.value, this.URI), { Referer: 'mangahere.com' }));
+        const pages = await Common.FetchPagesSinglePageJS.call(this, chapter, pageScript);
+        return pages.map(page => new Page(this, chapter, page.Link, { Referer: 'https://mangahere.com' }));
```

Comment:
Avoid intermediate mapping from `Page[]` to `Page[]` just to change a single property, instead get the pages directly.
```javascript
const data = await FetchWindowScript<string[]>(new Request(new URL(chapter.Identifier, this.URI)), pageScript);
return data.map(link => new Page(this, chapter, new URL(link), { Referer: 'https://mangahere.com' }));
```